### PR TITLE
Expose more cache behavior parameters.

### PR DESCRIPTION
**Pull request by Pavel910**
This PR is discussed in #6. It adds more behavior parameters to both default and custom cache behaviors.

Existing tests are passing as I left the existing defaults intact, and I also added a new test with a snapshot. This was also tested manually on the project I 'm working on.

`README.md` also updated to expand the examples with new parameters.

@eahefnawy please review and let me know if you want to change something!

### CODE REVIEW

**Review by josephluck — COMMENTED** [review]
Really nice work. Looks like this could be extended further to provide more configuration options for things like `TrustedSigners` etc.

This fixes a problem I've seen with the `serverless-next-js` plugin where cookies are not forwarded to edge lambdas: https://github.com/danielcondemarin/serverless-next.js/issues/154

**Comment by Pavel910** [comment]
@josephluck thanks! Headers/cookies forwarding is the primary reason I created this PR, as I need those in my own platform. I've personally never used `TrustedSigners` so far, so I skipped those, but other than that I think I've included all of the options documented in the AWS Docs.

File: README.md [hunk 1]
```diff
@@ -53,14 +53,25 @@ distribution:
...
+        # array of header names, 'none' or 'all'
+        headers: ['Accept', 'Accept-Language']
+        # array of cookie names, 'none' or 'all'
+        cookies: ['my-cookie]
+        queryString: true
+        queryStringCacheKeys: ['queryKey']
+      viewerProtocolPolicy: 'https-only' # optional
+      smoothStreaming: true # optional
+      compress: true # optional
+      fieldLevelEncryptionId: '123' # optional
```
Comment (eahefnawy) [inline README.md:66]: love that all is optional and we still have sane defaults 👍

**Review by eahefnawy — APPROVED** [review]
Thanks @Pavel910 ... looks great! 👍

**Pull request merged**